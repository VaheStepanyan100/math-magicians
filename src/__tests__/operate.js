import operate from '../logic/operate';

describe('Operate function test', () => {
  test('Performs addition correctly ', () => {
    const result = operate('3', '4', '+');
    expect(result).toMatchSnapshot();
  });
  test('Performs multiplication correctly ', () => {
    const result = operate('3', '4', 'x');
    expect(result).toMatchSnapshot();
  });
  test('Performs substraction correctly ', () => {
    const result = operate('3', '4', '-');
    expect(result).toMatchSnapshot();
  });
  test('Performs division correctly ', () => {
    const result = operate('3', '4', 'รท');
    expect(result).toMatchSnapshot();
  });

  test("return Can't divide by 0.", () => {
    const result = operate('60', '0', 'รท');
    expect(result).toEqual("Can't divide by 0.");
  });
});
