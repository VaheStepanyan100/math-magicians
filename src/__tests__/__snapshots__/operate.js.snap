// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operate function test Performs addition correctly  1`] = `"7"`;

exports[`Operate function test Performs division correctly  1`] = `"0.75"`;

exports[`Operate function test Performs multiplication correctly  1`] = `"12"`;

exports[`Operate function test Performs multiplication correctly : 12 1`] = `"12"`;

exports[`Operate function test Performs substraction correctly  1`] = `"-1"`;
